dandischema/models.py

```diff
@@ -57,12 +57,22 @@
     "[a-f0-9]{8}[-]*[a-f0-9]{4}[-]*" "[a-f0-9]{4}[-]*[a-f0-9]{4}[-]*[a-f0-9]{12}$"
 )
 ASSET_UUID_PATTERN = r"^dandiasset:" + UUID_PATTERN
-VERSION_PATTERN = r"\d{6}/\d+\.\d+\.\d+"
-DANDI_DOI_PATTERN = rf"^10.(48324|80507)/dandi\.{VERSION_PATTERN}"
+DANDI_ID_PATTERN = r"\d{6}"
+VERSION_PATTERN = rf"{DANDI_ID_PATTERN}/\d+\.\d+\.\d+"
+DANDI_DOI_WITH_VERSION = rf"^10.(48324|80507)/dandi\.{VERSION_PATTERN}"
+DANDI_DOI_NO_VERSION = r"^10\.(48324|80507)/dandi\.\d{6}"
+DANDI_DOI_PATTERN = rf"{DANDI_DOI_WITH_VERSION}|{DANDI_DOI_NO_VERSION}"
 DANDI_PUBID_PATTERN = rf"^DANDI:{VERSION_PATTERN}"
+
+PUBLISHED_DANDISET_URL_PATTERN = (
+    rf"^{DANDI_INSTANCE_URL_PATTERN}/dandiset/{DANDI_ID_PATTERN}"
+)
 PUBLISHED_VERSION_URL_PATTERN = (
     rf"^{DANDI_INSTANCE_URL_PATTERN}/dandiset/{VERSION_PATTERN}$"
 )
+PUBLISHED_URL_PATTERN = (
+    rf"{PUBLISHED_VERSION_URL_PATTERN}|{PUBLISHED_DANDISET_URL_PATTERN}"
+)
```

Comment:
TODO: this has gotten really messy, it can be cleaned up (i just hacked it out to work for now)

Reply:
Some of those will be outdated soon. We are modifying some of them in https://github.com/dandi/dandi-schema/pull/294. The goal for that PR is not to clean the mess though.

Reply:
@candleindark thanks for lookin out, nice catch!

Reply:
I wonder if the model changes are still necessary at all? I've gone through several implementations that were trying to pass all the data through Pydantic validation first, and we've also made the decision to store the dandiset-wide doi on the draft version. If we aren't going to soften the Dandiset pydantic model so we can use that for a draft dandiset, I'm not sure if the draft could ever pass PublishedDandiset validation. So there isnt really a need to change the regex to accept that other format of doi...